userservice/src/main/java/com/infodation/userservice/services/NotificationServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.infodation.userservice.services;
+
+import com.infodation.userservice.mapper.NotificationMapper;
+import com.infodation.userservice.models.User;
+import com.infodation.userservice.models.notimodel.CreateNotificationDTO;
+import com.infodation.userservice.models.notimodel.Notifications;
+import com.infodation.userservice.repositories.NotificationRepository;
+import com.infodation.userservice.repositories.UserRepository;
+import com.infodation.userservice.services.iservice.INotificationService;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class NotificationServiceImpl implements INotificationService {
+    private final NotificationRepository notiRepository;
+    private final UserRepository userRepository;
+
+    public NotificationServiceImpl(NotificationRepository notiRepository, UserRepository userRepository) {
+        this.notiRepository = notiRepository;
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public Notifications saveNoti(CreateNotificationDTO notiDTO) {
+        // Find the user by userId
+        Optional<User> userOptional = userRepository.findByUserId(notiDTO.getUserId());
+        if (userOptional.isEmpty()) {
+            throw new RuntimeException("User not found");
```

Comment:
It is good to define a custom exception `UserNotFoundException` and extend `Exception` for our application rather than using base `RuntimeException`, this will make the code more readable.